article/app/controllers/ArticleController.scala

```diff
@@ -180,128 +34,61 @@ class ArticleController(contentApiClient: ContentApiClient, val controllerCompon
-    }
-
-  }
-
   def renderArticle(path: String): Action[AnyContent] = {
-
     Action.async { implicit request =>
-
       if(request.isGuui){
-
-        timedFuture(
-          mapModelGUUI(path, Some(ArticleBlocks)){
-            remoteRender(path, _)
-          },
-          ArticleRenderingMetrics.RemoteRenderingMetric
-        )
-
+          mapModel(path, Some(ArticleBlocks)){
+             remoteRender.render(ws, path, _)
+          }
       } else {
-
-        timedFuture(
           mapModel(path, Some(ArticleBlocks)) {
             render(path, _)
-          },
-          ArticleRenderingMetrics.LocalRenderingMetric
-        )
-
+          }
       }
-
     }
   }
 
-  // range: None means the url didn't include /live/, Some(...) means it did.  Canonical just means no url parameter
-  // if we switch to using blocks instead of body for articles, then it no longer needs to be Optional
-  def mapModel(path: String, range: Option[BlockRange] = None, asArticle: Boolean = true)(render: PageWithStoryPackage => Result)(implicit request: RequestHeader): Future[Result] = {
-    lookup(path, range) map responseToModelOrResult(range, asArticle) recover convertApiExceptions map {
-      case Left(model) => render(model)
-      case Right(other) => RenderOtherStatus(other)
+  def renderEmail(path: String): Action[AnyContent] = {
+    Action.async { implicit request =>
+      mapModel(path, range = Some(ArticleBlocks)) {
+        render(path, _)
+      }
     }
   }
 
-  private def lookup(path: String, range: Option[BlockRange])(implicit request: RequestHeader): Future[ItemResponse] = {
-    val edition = Edition(request)
-
-    val capiItem = contentApiClient.item(path, edition)
-      .showTags("all")
-      .showFields("all")
-      .showReferences("all")
-      .showAtoms("all")
-
-    val capiItemWithBlocks = range.map { blockRange =>
-      val blocksParam = blockRange.query.map(_.mkString(",")).getOrElse("body")
-      capiItem.showBlocks(blocksParam)
-    }.getOrElse(capiItem)
-    contentApiClient.getResponse(capiItemWithBlocks)
+  private def render(path: String, page: PageWithStoryPackage)(implicit request: RequestHeader): Future[Result] = page match {
+    case article: ArticlePage =>
+      val htmlResponse = () => {
+        if (request.isEmail) ArticleEmailHtmlPage.html(article)
+        else if (request.isAmp) views.html.articleAMP(article)
+        else ArticleHtmlPage.html(article)
+      }
+      // add extra data for *.json?guui endpoint
+      val contentFieldsJson = if (request.isGuuiJson) List("contentFields" -> Json.toJson(ContentFields(article.article))) else List()
+      val jsonResponse = () => List(("html", views.html.fragments.articleBody(article))) ++ contentFieldsJson
+      Future { renderFormat(htmlResponse, jsonResponse, article) }
+  }
 
+  def mapModel(path: String, range: Option[BlockRange] = None)(render: PageWithStoryPackage => Future[Result])(implicit request: RequestHeader): Future[Result] = {
```

Comment:
You could change map model to take an `ArticlePage` to `Future[Result]` since we only care about ArticlePages in this class